File: itv/AMarotoChivite-ITV/AMarotoChivite-ITV-Java/src/main/java/FunctionalITVJava/models/Vehicle.java
package FunctionalITVJava.models;

public abstract class Vehicle {
    public final int anyo;
    public final int km;
    public final String brand;
    public final Boolean apto;

    Vehicle(String brand, int anyo, Boolean apto, int km) {
        this.brand = brand;
        this.anyo = anyo;
        this.apto = apto;
        this.km = km;
    }
}

File: itv/AMarotoChivite-ITV/AMarotoChivite-ITV-Kotlin/src/main/kotlin/FunctionalITV/models/Vehicle.kt
package FunctionalITV.models

sealed class Vehicle(
    val brand: String,
    val anyo: Int,
    val apto: Boolean,
    val km: Int
)

class Moto(
    brand: String,
    anyo: Int,
    apto: Boolean,
    km: Int,
    val cilindra: Int
) : Vehicle(brand, anyo, apto, km) {
    override fun toString(): String {
        return "Moto(brand='$brand', anyo=$anyo, apto=$apto, km=$km, cilindra=$cilindra)"
    }
}

class Car(
    brand: String,
    anyo: Int,
    apto: Boolean,
    km: Int,
    val numDoors: Int
) : Vehicle(brand, anyo, apto, km) {
    override fun toString(): String {
        return "Car(brand='$brand', anyo=$anyo, apto=$apto, km=$km, numDoors=$numDoors)"
    }
}